src/services/issue_analysis_service.py

```diff
@@ -0,0 +1,347 @@
+"""
+IssueAnalysisService - Responsible for issue analysis workflows.
+Handles filtering known issues, analyzing issues, and generating recommendations.
+"""
+
+import os
+import logging
+from typing import Tuple
+from langchain_core.prompts import ChatPromptTemplate, SystemMessagePromptTemplate, HumanMessagePromptTemplate
+from langchain_core.runnables import RunnablePassthrough, RunnableLambda
+from langchain_core.language_models.chat_models import BaseChatModel
+from langchain_community.vectorstores import FAISS
+from tenacity import retry, stop_after_attempt, wait_fixed, retry_if_exception_type
+
+from common.constants import FALLBACK_JUSTIFICATION_MESSAGE, RED_ERROR_FOR_LLM_REQUEST
+from dto.Issue import Issue
+from dto.ResponseStructures import FilterResponse, JudgeLLMResponse, JustificationsSummary, RecommendationsResponse, EvaluationResponse
+from dto.LLMResponse import AnalysisResponse
+from Utils.file_utils import read_answer_template_file
+from Utils.llm_utils import robust_structured_output
+from .vector_store_service import VectorStoreService
+
+logger = logging.getLogger(__name__)
+
+
+def _format_context_from_response(resp):
```

Comment:
Where do we use this function? It appears to me, this one will more fit into `vector_store_service.py`

Reply:
its a left over from the rebase (it was used in the old `filter_known_issues` function), and its not being called anymore